src/Statistics/include/Statistics/Histogram.h

```diff
@@ -0,0 +1,33 @@
+// Copyright (c) 2022 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef STATISTICS_HISTOGRAM_H_
+#define STATISTICS_HISTOGRAM_H_
+
+#include <cstdint>
+#include <optional>
+#include <vector>
+
+namespace orbit_statistics {
+
+// A histogram of a dataset of `uint64_t` values with bins of equal width (with a possible
+// exception for the last one). The bins are:
+// [min, min + bin_width), [min + bin_width, min + 2*bin_width), ...
+// ... [min + (counts*size() - 1) * bin_width, max]
+// `counts[i]` stores the number of elements in i-th bin.
+struct Histogram {
+  uint64_t min{};
+  uint64_t max{};
+  uint64_t bin_width{};
+  size_t data_set_size{};
+  std::vector<size_t> counts;
+};
+
+// The function builds multiple histograms with different number of bins,
+// estimates the risk score using `HistogramRiskScore` and returns the histogram
+// which minimizes it.
+[[nodiscard]] std::optional<Histogram> BuildHistogram(const std::vector<uint64_t>& data);
```

Comment:
I would suggest renaming this to distinguish from the one in `HistogramUtils`. Something like `BuildBestHistogram` (although I don't know if "best" is ideal).

Reply:
I'd suggest the word "optimal", but it's not optimal either, as we only run through a finite grid. The correct and informative name for the function would be `BuildHistogramWithAutoBinWidth`. 

I'd prefer to keep the `BuildHistogram` name, as the difference between the two overloads is clear -- the caller either passes the bin_width, or the function is expected to choose bin_width automatically.